services/provider/server/server.go

```diff
@@ -878,6 +895,36 @@ func (s *OCSProviderServer) getOCSSubscriptionChannel(ctx context.Context) (stri
 	return subscription.Spec.Channel, nil
 }
 
+func (s *OCSProviderServer) getStorageCluster(ctx context.Context) (*ocsv1.StorageCluster, error) {
```

Comment:
the implementation seems to be a roundabout way to me, in a namespace there'll only be 1 storagecluster isn't it? you could limit the list by 2 and anything that isn't 1 is an error scenario.

moreover from 4.18 api-server is always deployed and checking storagecluster for external enabled, the condition is implicitly missing out internal only scenario and I suggest storagecluster w/ allowremotestorageclusters atm is more suitable for the condition here.

Reply:
Acc to how the current implimenetation is if there are multiple storageClusters, only one of them is handled, the others are just moved into Phase: Ignored. So I don't think erroring out ACC to the lits's size would be a good indicator.  

Agree, specifically looking for cluster with allowremotestorageclusters seems to be the best filter.

Reply:
Updated the func

Reply:
> into Phase: Ignored
- didn't know that, thanks!

> I don't think erroring out ACC to the lits's size would be a good choice.
- fair enough, then you should also be filtering out storagecluster that is set w/ ignored phase isn't it?

Reply:
Right, Added a check for that.